agc/anal.c

```diff
@@ -0,0 +1,116 @@
+
+        // control flow and interrupts
+        case AGC_INSN_TCF: // jump to fixed memory
+            return R_ANAL_OP_TYPE_JMP;
+
+        case AGC_INSN_OVSK: // skip next instruction on overflow
+        case AGC_INSN_CCS: // conditional jump to erasable memory
+        case AGC_INSN_BZF: // branch to fixed memory if `A` = 0
+        case AGC_INSN_BZMF: // branch to fixed memory if `A` <= 0
+            return R_ANAL_OP_TYPE_CJMP;
+
+        case AGC_INSN_TCAA: // jump to address in `A`
+        case AGC_INSN_XLQ: // jump to `A`
+        case AGC_INSN_XXALQ: // jump to `A` with extracode
+            return R_ANAL_OP_TYPE_UCJMP;
+
+        case AGC_INSN_TC: // call subroutine
+            return R_ANAL_OP_TYPE_CALL;
+
+        case AGC_INSN_RETURN: // return from subroutine
+            return R_ANAL_OP_TYPE_RET;
+
+        case AGC_INSN_INHINT: // diable interrupts
+        case AGC_INSN_RELINT: // enable interrupts
+        case AGC_INSN_RESUME: // resume after an ISR
+        case AGC_INSN_EXTEND: // extend next instruction
+                              // (not directly interrupt-related)
+            return R_ANAL_OP_TYPE_SWI;
+
+        // moving memory
+        case AGC_INSN_TS: // transfer to storage and manipulate `A`
+            return R_ANAL_OP_TYPE_CMOV; // TODO: maybe make it STORE?
+
+        case AGC_INSN_CA: // move memory into the `A` register
+        case AGC_INSN_DCA: // move memory into `A`, `L` register pair
+            return R_ANAL_OP_TYPE_LOAD;
+
+        case AGC_INSN_DTCB: // memory adressing change using a register swap
+        case AGC_INSN_DTCF: // similar to the above
+        case AGC_INSN_ZL: // clear `L` register
+        case AGC_INSN_ZQ: // clear `Q` register
+        case AGC_INSN_LXCH: // exchange with `L` register
+        case AGC_INSN_DXCH: // exchange with `A`, `L` register pair
+        case AGC_INSN_XCH: // exchange with `A` register
+        case AGC_INSN_QXCH: // exchange with `Q` register
+            return R_ANAL_OP_TYPE_XCHG;
+
+        // special instructions and I/O
+        case AGC_INSN_NOOP: // do nothing
+            return R_ANAL_OP_TYPE_NOP;
+
+        case AGC_INSN_READ: // read IO channel into `A`
+        case AGC_INSN_WRITE: // write contents of `A` into IO channel
+        case AGC_INSN_RAND: // read IO channel into `A`, with and
+        case AGC_INSN_WAND: // write contents of `A` into IO channel, with and
+        case AGC_INSN_ROR: // read IO channel into `A`, with or
+        case AGC_INSN_WOR: // write contents of `A` into IO channel, with or
+        case AGC_INSN_RXOR: // read IO channel into `A`, with xor
+            return R_ANAL_OP_TYPE_IO;
+
+
+        case AGC_INSN_EDRUPT: // we don't know what it does
+        default:
+            return R_ANAL_OP_TYPE_UNK;
+    }
+}
+
+void analyze_agc_insn(
+        RAnalOp *op, unsigned int address, ut16 value, bool shift) {
+
+    agc_insn_t insn = {0};
+    disasm_agc_insn(&insn, address, value, shift);
```

Comment:
dont mix tabs and spaces. use only tabs for indentation.

missing space before (